src/webportal/src/app/job-submission/components/tools/ssh-generator.jsx

```diff
@@ -0,0 +1,218 @@
+// The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING
+// BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+// NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+// DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+import React, { useEffect, useState } from 'react';
+import { Hint } from '../sidebar/hint';
+import {
+  DefaultButton,
+  Dialog,
+  DialogType,
+  FontClassNames,
+  FontSizes,
+  Label,
+  Separator,
+  Stack,
+  TextField,
+} from 'office-ui-fabric-react';
+import PropTypes from 'prop-types';
+import c from 'classnames';
+import t from '../../../components/tachyons.scss';
+import { generateSSHKeyPair } from '../../utils/ssh-keygen';
+import { isNil } from 'lodash';
+
+export default function SSHGenerator({
+  isOpen = false,
+  hide,
+  onSshKeysChange,
+}) {
+  const [sshKeys, setSshKeys] = useState();
+  const [downloadedPriKey, setDownloadedPriKey] = useState(false);
+
+  useEffect(() => {
+    generateSshKeys(1024);
+  }, []);
+
+  useEffect(() => {
+    setDownloadedPriKey(false);
+  }, [sshKeys]);
+
+  const generateSshKeys = (bits, ev = undefined) => {
+    const keys = generateSSHKeyPair(bits);
+    setSshKeys(keys);
+  };
+
+  const handleCancel = () => {
+    hide();
+  };
+
+  const handleConfirm = () => {
+    onSshKeysChange(sshKeys);
+    hide();
+  };
+
+  const downloadAsFile = (content, saveName, ev) => {
+    const blob = new Blob([content], {
+      type: 'application/octet-stream,charset=UTF-8',
+    });
+    openDownloadDialog(blob, saveName);
+  };
+
+  const openDownloadDialog = (url, saveName) => {
+    if (typeof url === 'object' && url instanceof Blob) {
+      url = URL.createObjectURL(url);
+    }
+    const aLink = document.createElement('a');
+    aLink.href = url;
+    aLink.download = saveName || '';
+    let event;
+    if (window.MouseEvent) event = new MouseEvent('click');
+    else {
+      event = document.createEvent('MouseEvents');
+      event.initMouseEvent(
+        'click',
+        true,
+        false,
+        window,
+        0,
+        0,
+        0,
+        0,
+        0,
+        false,
+        false,
+        false,
+        false,
+        0,
+        null,
+      );
+    }
+    aLink.dispatchEvent(event);
+  };
+
+  return (
+    <Dialog
+      hidden={!isOpen}
+      onDismiss={handleCancel}
+      dialogContentProps={{
+        type: DialogType.normal,
+        styles: {
+          title: { padding: '20px 36px 12px 20px' },
+          inner: { padding: '0px 40px 20px 20px' },
+          topButton: { padding: '20px 20px 0px 0px' },
```

Comment:
I think we'd better use the spacing values in fabric's theme if possible.

Reply:
Removed